[dynamo+aten] Enable embedding_bag_byte_unpack + meta kernel impl

hl475 opened the pull request:
Summary:
```
torch._dynamo.exc.Unsupported: unsupported operator: quantized.embedding_bag_byte_unpack.default
```

Differential Revision: D48652953



cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

houseroad commented on test/test_meta.py:
```diff
@@ -1303,6 +1303,32 @@ def test_map_location_deserialize(self):
+        for ed, rs in zip(embedding_dim, res_shape):
+            packed_weight = torch.randn(batch_size, num_embeddings, ed + 8, dtype=torch.float32)
+            res = torch.ops.quantized.embedding_bag_byte_unpack(packed_weight.to(device="meta"))
+            print(res.shape)
```
nit: remove print?

houseroad commented on test/test_meta.py:
```diff
@@ -1303,6 +1303,32 @@ def test_map_location_deserialize(self):
+        for ed, rs in zip(embedding_dim, res_shape):
+            weight = torch.randn(batch_size, num_embeddings, ed, dtype=torch.float32)
+            res = torch.ops.quantized.embedding_bag_byte_prepack(weight.to(device="meta"))
+            print(res.shape)
```
remove print

houseroad commented on test/test_meta.py:
```diff
@@ -1303,6 +1303,32 @@ def test_map_location_deserialize(self):
+        num_embeddings = 80
+        embedding_dim = [128, 256, 512]
+        res_shape = [[batch_size, num_embeddings, ed + 8] for ed in embedding_dim]
+        for ed, rs in zip(embedding_dim, res_shape):
```
a better test approach is using the cpu backend to run and compare the shape.

hl475 replied:
lol I tried that in this test_meta.py, but it does not allow me to use cpu backend. Instead, I used the op bench for two backends to compare the shape.

houseroad reviewed (approved):
Looks good.

hl475 committed:
[dynamo+aten] Enable embedding_bag_byte_unpack + meta kernel impl (#107937)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/107937

```
torch._dynamo.exc.Unsupported: unsupported operator: quantized.embedding_bag_byte_unpack.default
```

Differential Revision: D48652953

fbshipit-source-id: 102a76468addbeb1fbc7de90ffccc62e8fd3400e

hl475 commented:
@pytorchbot merge
